### DISCUSSION TITLE
Is there a way to set project global configuration?

### DISCUSSION THREAD

**Discussion by eproxus** [pulumi/pulumi#8731] [Pulumi in Practice] [answered]
When using `pulumi config set foo bar` the configuration ends up in the current stack configuration. Is there a way to set project global configuration that is accessible from all stacks?

My use case is, I want to deploy servers with a list of SSH public keys and I want to re-use the list of public keys for all stacks.

**Comment by Frassle** [comment, marked as answer]
We have an issue https://github.com/pulumi/pulumi/issues/2307 to look into supporting per-project and per-organisation config. The current suggested workaround is making a stack just for config and using stack references to refer to it.

**Comment by cnunciato** [comment]
Stack references docs are here: https://www.pulumi.com/docs/intro/concepts/stack/#stackreferences

**Comment by rawkode** [comment]
Hey @eproxus,

Fraser is correct, however it's worth noting that there's some current workarounds for this situation. 

You could store a JSON or YAML list of GitHub users at some URI and consume it with your Pulumi code. This would allow you to use GitHubs magic `https://GitHub.com/rawkode.keys` endpoint to grab SSH keys for users. 

Of course you could also just store a list of approved keys somewhere.

If you're doing this in JS or TS, I'd encourage using a dynamic provider for this. If you'd like an example, let me know and I'll put something together. 

Best of luck
